### ISSUE TITLE
Duplicate events being published?

### ISSUE THREAD

**Issue by mveitas** [honeycombio/honeycomb-kubernetes-agent#146]
I was running the K8S agent on an EKS cluster as a way to explore resources usage and noticed that there were a number of events that were published to Honeycomb with the same timestamps, the only difference was the uptime_status value.

I wanted to open up an issue around this while I explore it a bit more.

```
Timestamp,k8s.cluster.name,k8s.namespace.name,k8s.pod.name,metrics.cpu.usage,metrics.cpu.utilization,metrics.source,metrics.source.type,metrics.uptime,status.phase,status.reason
2021-03-26T18:24:48Z,k8s-cluster,datadog,datadog-hgm4c,0.214861,61.388803,datadog-hgm4c,pod,1489820362.000000,Running,
2021-03-26T18:24:48Z,k8s-cluster,datadog,datadog-hgm4c,0.214861,61.388803,datadog-hgm4c,pod,1489830336.000000,Running,
2021-03-26T18:24:34Z,k8s-cluster,datadog,datadog-hgm4c,0.215177,61.479246,datadog-hgm4c,pod,1489810326.000000,Running,
2021-03-26T18:24:20Z,k8s-cluster,datadog,datadog-hgm4c,0.222227,63.493410,datadog-hgm4c,pod,1489790351.000000,Running,
2021-03-26T18:24:20Z,k8s-cluster,datadog,datadog-hgm4c,0.222227,63.493410,datadog-hgm4c,pod,1489800349.000000,Running,
2021-03-26T18:24:04Z,k8s-cluster,datadog,datadog-hgm4c,0.222834,63.666980,datadog-hgm4c,pod,1489780329.000000,Running,
2021-03-26T18:23:48Z,k8s-cluster,datadog,datadog-hgm4c,0.216643,61.897915,datadog-hgm4c,pod,1489770326.000000,Running,
2021-03-26T18:23:48Z,k8s-cluster,datadog,datadog-hgm4c,0.216643,61.897915,datadog-hgm4c,pod,1489760347.000000,Running,
2021-03-26T18:23:37Z,k8s-cluster,datadog,datadog-hgm4c,0.213209,60.916734,datadog-hgm4c,pod,1489750363.000000,Running,
2021-03-26T18:23:23Z,k8s-cluster,datadog,datadog-hgm4c,0.225316,64.375927,datadog-hgm4c,pod,1489740364.000000,Running,
2021-03-26T18:23:08Z,k8s-cluster,datadog,datadog-hgm4c,0.213428,60.979339,datadog-hgm4c,pod,1489720348.000000,Running,
2021-03-26T18:23:08Z,k8s-cluster,datadog,datadog-hgm4c,0.213428,60.979339,datadog-hgm4c,pod,1489730353.000000,Running,
2021-03-26T18:22:53Z,k8s-cluster,datadog,datadog-hgm4c,0.217917,62.261910,datadog-hgm4c,pod,1489710343.000000,Running,
2021-03-26T18:22:41Z,k8s-cluster,datadog,datadog-hgm4c,0.202066,57.733284,datadog-hgm4c,pod,1489700350.000000,Running,
2021-03-26T18:22:22Z,k8s-cluster,datadog,datadog-hgm4c,0.217025,62.007042,datadog-hgm4c,pod,1489680338.000000,Running,
2021-03-26T18:22:22Z,k8s-cluster,datadog,datadog-hgm4c,0.217025,62.007042,datadog-hgm4c,pod,1489690325.000000,Running,
2021-03-26T18:22:11Z,k8s-cluster,datadog,datadog-hgm4c,0.118579,33.879611,datadog-hgm4c,pod,1489660339.000000,Running,
2021-03-26T18:22:11Z,k8s-cluster,datadog,datadog-hgm4c,0.118579,33.879611,datadog-hgm4c,pod,1489670353.000000,Running,
2021-03-26T18:21:54Z,k8s-cluster,datadog,datadog-hgm4c,0.219662,62.760693,datadog-hgm4c,pod,1489650354.000000,Running,
2021-03-26T18:21:37Z,k8s-cluster,datadog,datadog-hgm4c,0.219561,62.731762,datadog-hgm4c,pod,1489630334.000000,Running,
2021-03-26T18:21:37Z,k8s-cluster,datadog,datadog-hgm4c,0.219561,62.731762,datadog-hgm4c,pod,1489640333.000000,Running,
2021-03-26T18:21:24Z,k8s-cluster,datadog,datadog-hgm4c,0.219794,62.798323,datadog-hgm4c,pod,1489620356.000000,Running,
2021-03-26T18:21:06Z,k8s-cluster,datadog,datadog-hgm4c,0.203359,58.102659,datadog-hgm4c,pod,1489610349.000000,Running,
2021-03-26T18:21:06Z,k8s-cluster,datadog,datadog-hgm4c,0.203359,58.102659,datadog-hgm4c,pod,1489600325.000000,Running,
2021-03-26T18:20:52Z,k8s-cluster,datadog,datadog-hgm4c,0.217406,62.115971,datadog-hgm4c,pod,1489590356.000000,Running,
```

**Comment by mveitas** [comment]
I spent some time looking into this and what I found was that at the default collection interval of 10s there appear to be perceived duplicates. The agent appears to be doing the correct thing and pulling the stats that come from kublet for the resources every 10s, but the underlying resources aren't being updated and hence why we get some perceived duplicate values. I am not sure how often the kublet updates the stats and probably correlated to the issue I was seeing.  (note - I haven't dug into how often the underlying stats are collected by kublet and assume this is something that is configurable)

Running a local minikube cluster and increasing the metrics collection interval from 10s to 15s and "duplicates" are reduced to be very infrequent. I am going to run a little test to find out the duplicate rate using a range of 10s to 15s intervals (will add a new field with the resource timestamp to allow me to do a group by)

For clusters that have 100s of resources, these "duplicate" events can start to add up and consume the events for a given team.
